lib/logstash/outputs/s3.rb

```diff
@@ -79,6 +79,8 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
   config_name "s3"
   default :codec, 'line'
 
+  concurrency :single
```

Comment:
humm the PR name is "new concurrency model :shared" and the description says the plugin threadsafe, however this sets the model to `:single`? this means that 1 instance is shared among all threads and access to it is globally synchronized.